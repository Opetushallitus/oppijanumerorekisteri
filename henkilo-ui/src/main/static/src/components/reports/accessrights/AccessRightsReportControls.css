.access-right-report-controls-row {
    padding-top: 0.5em;
}
