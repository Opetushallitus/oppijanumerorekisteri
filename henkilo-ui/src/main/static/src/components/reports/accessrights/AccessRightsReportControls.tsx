import React from 'react';
import type { Option, Options } from 'react-select';
import OrganisaatioSelectModal from '../../common/select/OrganisaatioSelectModal';
import ItemList from '../../kayttooikeusryhmat/kayttooikeusryhma/ItemList';
import OphSelect from '../../common/select/OphSelect';
import Button from '../../common/button/Button';
import DownloadIcon from '../../common/icons/DownloadIcon';
import { OrganisaatioSelectObject } from '../../../types/organisaatioselectobject.types';
import './AccessRightsReportControls.css';
import { useLocalisations } from '../../../selectors';

type Props = {
    disabled: boolean;
    filterValues: string[];
    filter: string;
    setFilter: (name: string) => void;
    setOid: (oid: string) => void;
    dataExport?: () => void;
};

const AccessRightsReportControls: React.FC<Props> = ({
    disabled,
    setOid,
    filter,
    filterValues,
    setFilter,
    dataExport,
}) => {
    const [selectedOrganisation, setSelectedOrganisation] = React.useState<OrganisaatioSelectObject[]>([]);
    const onSelect = (organisaatio: OrganisaatioSelectObject) => setSelectedOrganisation([organisaatio]);
    const { L } = useLocalisations();

    React.useEffect(() => {
        setOid(selectedOrganisation[0] && selectedOrganisation[0].oid);
    }, [setOid, selectedOrganisation]);

    const filterOptions: Options<string> = filterValues.map((name) => ({ label: name, value: name }));

    return (
        <div>
            <div className="flex-horizontal">
                <div className="flex-item-1 ">
                    <OrganisaatioSelectModal onSelect={onSelect} disabled={disabled} />
                    <ItemList
                        items={selectedOrganisation}
                        labelPath={['name']}
                        removeAction={() => setSelectedOrganisation([])}
                    />
                </div>
            </div>
            {filterOptions.length > 1 && (
                <div className="flex-horizontal access-right-report-controls-row">
                    <div className="flex-item-1 ">
                        <OphSelect
                            options={filterOptions}
                            placeholder={L['HENKILOHAKU_FILTERS_KAYTTOOIKEUSRYHMA_PLACEHOLDER']}
                            value={filter}
                            clearable
                            onChange={(option: Option<string>) => setFilter(option && option.value)}
                        />
                    </div>
                </div>
            )}
            {dataExport && (
                <div className="flex-horizontal access-right-report-controls-row">
                    <div className="flex-item-1 ">
                        <Button action={dataExport}>
                            {L['KAYTTOOIKEUSRAPORTTI_EXPORT']}
                            <DownloadIcon />
                        </Button>
                    </div>
                </div>
            )}
        </div>
    );
};

export default AccessRightsReportControls;
