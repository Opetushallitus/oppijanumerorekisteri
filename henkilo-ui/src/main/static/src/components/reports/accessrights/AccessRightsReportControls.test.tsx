import { shallow } from 'enzyme';
import AccessRightsReportControls from './AccessRightsReportControls';

describe('AccessRightsReportControls', () => {
    const MINIMAL_PROPS = {
        locale: '',
        L: {},
        organisations: [],
        disabled: false,
        filterValues: [],
        filter: undefined,
        setFilter: jest.fn(),
        setOid: jest.fn(),
    };

    test('renders without crashing', () => {
        expect(shallow(<AccessRightsReportControls {...MINIMAL_PROPS} />)).toMatchSnapshot();
    });

    test('invokes setOid callback in useEffect', () => {
        shallow(<AccessRightsReportControls {...MINIMAL_PROPS} />);
        expect(MINIMAL_PROPS.setOid).toHaveBeenCalledWith(undefined);
    });

    test('renders access rights filter if more than one access rights', () => {
        const component = shallow(<AccessRightsReportControls {...MINIMAL_PROPS} />);
        expect(component.find('Connect(OphSelect)')).toHaveLength(0);
        component.setProps({ ...MINIMAL_PROPS, filterValues: ['foo'] });
        expect(component.find('Connect(OphSelect)')).toHaveLength(0);
        component.setProps({ ...MINIMAL_PROPS, filterValues: ['foo', 'bar'] });
        expect(component.find('Connect(OphSelect)')).toHaveLength(1);
    });

    test('renders export button only if there is something to export', () => {
        const callback = jest.fn();
        const component = shallow(<AccessRightsReportControls {...MINIMAL_PROPS} />);
        expect(component.find('Button')).toHaveLength(0);
        component.setProps({ ...MINIMAL_PROPS, dataExport: callback });
        expect(component.find('Button')).toHaveLength(1);
    });
});
