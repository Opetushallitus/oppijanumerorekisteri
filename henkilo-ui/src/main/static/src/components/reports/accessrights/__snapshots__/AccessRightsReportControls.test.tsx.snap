// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessRightsReportControls renders without crashing 1`] = `
<div>
  <div
    className="flex-horizontal"
  >
    <div
      className="flex-item-1 "
    >
      <OrganisaatioSelectModal
        L={Object {}}
        disabled={false}
        locale=""
        onSelect={[Function]}
        organisaatiot={Array []}
      />
      <Connect(ItemList)
        items={Array []}
        labelPath={
          Array [
            "name",
          ]
        }
        removeAction={[Function]}
      />
    </div>
  </div>
</div>
`;
